test/unittests/testInit.cpp

```diff
@@ -112,18 +112,22 @@ static bool equalContextAttribute(ContextAttribute* caExpected, ContextAttribute
     LM_M(("caArg '%s', '%s', '%s'",
           caArg->name.c_str(),
           caArg->type.c_str(),
-          caArg->value.c_str()
+          caArg->toStringValue().c_str()
           ));
     LM_M(("caExpected '%s', '%s', '%s'",
           caExpected->name.c_str(),
           caExpected->type.c_str(),
-          caExpected->value.c_str()
+          caExpected->toStringValue().c_str()
           ));
 
-    if (caArg->name == caExpected->name && caArg->type == caExpected->type && caArg->value == caExpected->value) {
+    if (caArg->name == caExpected->name &&
+        caArg->type == caExpected->type &&
+        caArg->stringValue == caExpected->stringValue) // FIXME P10: automacit value -> stringValue change, please review to check if it is safe
```

Comment:
This fixme here seems adequate ...

It assumes the valueType to be of "String" type. And caExpected also.
What if you call this function with two attributes with Bool valurType?
Then two empty strings will be compared and it will say "OK".

I think you need to make the comparison a bit more complex:

    if (caArg->name != caExpected->name)     return false;
    if (caArg->type != caExpected->type)          return false;

    if (caArg->valueType != caExpected->valueType)    return false;

    if ((caArg->valueType == Bool) && (caArg->boolValue != caExpected->boolValue))   return false;

    etc.